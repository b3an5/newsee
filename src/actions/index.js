export const saveTopStories = article => ({
  type: 'SAVE_TOP_STORIES',
  article,
})

export const saveSearchStories = article => ({
  type: 'SAVE_SEARCH_STORIES',
  article,
})

export const saveCurrentStory = article => ({
  type: 'SAVE_CURRENT_STORY',
  article,
})
