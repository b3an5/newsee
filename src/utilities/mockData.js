export const mockCurrentStory = {
  id: null,
  name: 'Npr.org',
  author: '',
  title:
    "What It's Like To Live With A Foot In China, Another In The U.S. - NPR",
  description:
    "In a special series, Morning Edition discovers the experiences of people affected by the deepening tensions between the world's two largest economies.",
  url:
    'https://www.npr.org/2019/06/04/727829423/what-its-like-to-live-with-a-foot-in-china-another-in-the-u-s',
  urlToImage:
    'https://media.npr.org/assets/img/2019/06/03/china-two-worlds_wide-fcccc59b7e253b60deadfbdfaeb76e9f5de09a5c.jpg?s=1400',
  publishedAt: '2019-06-04T16:13:34Z',
  content:
    'American flags are displayed together with Chinese flags on top of a three-wheeled rickshaw in Beijing in 2018',
  source: { id: 'blah' },
}

export const mockTopStories = [
  {
    id: null,
    name: 'Npr.org',
    author: '',
    title:
      "What It's Like To Live With A Foot In China, Another In The U.S. - NPR",
    description:
      "In a special series, Morning Edition discovers the experiences of people affected by the deepening tensions between the world's two largest economies.",
    url:
      'https://www.npr.org/2019/06/04/727829423/what-its-like-to-live-with-a-foot-in-china-another-in-the-u-s',
    urlToImage:
      'https://media.npr.org/assets/img/2019/06/03/china-two-worlds_wide-fcccc59b7e253b60deadfbdfaeb76e9f5de09a5c.jpg?s=1400',
    publishedAt: '2019-06-04T16:13:34Z',
    content:
      'American flags are displayed together with Chinese flags on top of a three-wheeled rickshaw in Beijing in 2018',
    source: { id: 'blah' },
  },
  {
    id: null,
    name: 'Npr.org',
    author: '',
    title:
      "What It's Like To Live With A Foot In China, Another In The U.S. - NPR",
    description:
      "In a special series, Morning Edition discovers the experiences of people affected by the deepening tensions between the world's two largest economies.",
    url:
      'https://www.npr.org/2019/06/04/727829423/what-its-like-to-live-with-a-foot-in-china-another-in-the-u-s',
    urlToImage:
      'https://media.npr.org/assets/img/2019/06/03/china-two-worlds_wide-fcccc59b7e253b60deadfbdfaeb76e9f5de09a5c.jpg?s=1400',
    publishedAt: '2019-06-04T16:13:34Z',
    content:
      'American flags are displayed together with Chinese flags on top of a three-wheeled rickshaw in Beijing in 2018',
    source: { id: 'blah' },
  },
]

export const mockSearchStories = [
  {
    id: null,
    name: 'Npr.org',
    author: '',
    title:
      "What It's Like To Live With A Foot In China, Another In The U.S. - NPR",
    description:
      "In a special series, Morning Edition discovers the experiences of people affected by the deepening tensions between the world's two largest economies.",
    url:
      'https://www.npr.org/2019/06/04/727829423/what-its-like-to-live-with-a-foot-in-china-another-in-the-u-s',
    urlToImage:
      'https://media.npr.org/assets/img/2019/06/03/china-two-worlds_wide-fcccc59b7e253b60deadfbdfaeb76e9f5de09a5c.jpg?s=1400',
    publishedAt: '2019-06-04T16:13:34Z',
    content:
      'American flags are displayed together with Chinese flags on top of a three-wheeled rickshaw in Beijing in 2018',
    source: { id: 'blah' },
  },
  {
    id: null,
    name: 'Npr.org',
    author: '',
    title:
      "What It's Like To Live With A Foot In China, Another In The U.S. - NPR",
    description:
      "In a special series, Morning Edition discovers the experiences of people affected by the deepening tensions between the world's two largest economies.",
    url:
      'https://www.npr.org/2019/06/04/727829423/what-its-like-to-live-with-a-foot-in-china-another-in-the-u-s',
    urlToImage:
      'https://media.npr.org/assets/img/2019/06/03/china-two-worlds_wide-fcccc59b7e253b60deadfbdfaeb76e9f5de09a5c.jpg?s=1400',
    publishedAt: '2019-06-04T16:13:34Z',
    content:
      'American flags are displayed together with Chinese flags on top of a three-wheeled rickshaw in Beijing in 2018',
    source: { id: 'blah' },
  },
]
