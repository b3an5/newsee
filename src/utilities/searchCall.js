import { apiKey } from "./apiKey";

export const searchData = async search => {
  var url =
    "https://newsapi.org/v2/everything?" +
    "q=" +
    search +
    "&" +
    "language=en&" +
    "sortBy=relevancy&" +
    "apiKey=" +
    apiKey;
  const response = await fetch(url);
  const result = await response.json();
  return result.articles;
};
