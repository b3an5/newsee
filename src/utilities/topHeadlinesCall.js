import { apiKey } from "./apiKey";

export const topHeadlineData = async () => {
  var url =
    "https://newsapi.org/v2/top-headlines?" +
    "country=us&" +
    "apiKey=" +
    apiKey;
  const response = await fetch(url);
  const result = await response.json();
  return result.articles;
};
