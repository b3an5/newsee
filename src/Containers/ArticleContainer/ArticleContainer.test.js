import React from 'react'
import { shallow } from 'enzyme'
import { apiKey } from '../../utilities/apiKey'
import { saveTopStories, saveSearchStories } from '../../actions/index'
import {
  mockCurrentStory,
  mockTopStories,
  mockSearchStories,
} from '../../utilities/mockData'
import {
  mapDispatchToProps,
  ArticleContainer,
  mapStateToProps,
} from './ArticleContainer'

describe('ArticleContainer', () => {
  describe('articleContainer component', () => {
    let wrapper

    beforeEach(() => {
      wrapper = shallow(
        <ArticleContainer
          topStories={mockTopStories}
          searchStories={mockSearchStories}
        />
      )
      window.fetch = jest.fn().mockImplementation(() =>
        // eslint-disable-next-line no-undef
        Promise.resolve({
          ok: true,
          // eslint-disable-next-line no-undef
          json: () => Promise.resolve(mockSearchStories),
        })
      )
    })

    it('should match default snapshot', () => {
      expect(wrapper).toMatchSnapshot()
    })

    it('should match snapshot when loading', () => {
      wrapper = shallow(
        <ArticleContainer topStories={[]} searchStories={mockSearchStories} />
      )
      expect(wrapper).toMatchSnapshot()
    })

    it('should match snapshot when searched', () => {
      wrapper
        .find('.search-input')
        .simulate('change', { target: { value: 'blah' } })
      wrapper.find('.search-form').simulate('submit', {
        preventDefault: () => {},
      })
      expect(wrapper).toMatchSnapshot()
    })

    it('should match snapshot when searched retrieves nothing', () => {
      wrapper = shallow(
        <ArticleContainer topStories={mockTopStories} searchStories={[]} />
      )
      wrapper
        .find('.search-input')
        .simulate('change', { target: { value: 'blah' } })
      wrapper.find('.search-form').simulate('submit', {
        preventDefault: () => {},
      })
      expect(wrapper).toMatchSnapshot()
    })

    it('should have the correct default state', () => {
      expect(wrapper.state()).toEqual({
        search: '',
        searched: false,
        favorites: false,
        errored: false,
      })
    })

    it('should change the state of favorites when favorite button is pressed', () => {
      expect(wrapper.state()).toEqual({
        search: '',
        searched: false,
        favorites: false,
        errored: false,
      })
      localStorage.favorites = JSON.stringify(mockTopStories)
      wrapper.find('.fav-button').simulate('click')
      expect(wrapper.state()).toEqual({
        search: '',
        searched: false,
        favorites: true,
        errored: false,
      })
    })

    it('should match snapshot when favorites is clicked', () => {
      expect(wrapper.state()).toEqual({
        search: '',
        searched: false,
        favorites: false,
        errored: false,
      })
      localStorage.favorites = JSON.stringify(mockTopStories)
      wrapper.find('.fav-button').simulate('click')
      expect(wrapper).toMatchSnapshot()
    })

    it('should change the state of search when its typed into', () => {
      expect(wrapper.state()).toEqual({
        search: '',
        searched: false,
        favorites: false,
        errored: false,
      })
      wrapper
        .find('.search-input')
        .simulate('change', { target: { value: 'blah' } })
      expect(wrapper.state()).toEqual({
        search: 'blah',
        searched: false,
        favorites: false,
        errored: false,
      })
    })

    it('should search when searchform is submitted', () => {
      wrapper
        .find('.search-input')
        .simulate('change', { target: { value: 'blah' } })
      wrapper.find('.search-form').simulate('submit', {
        preventDefault: () => {},
      })
      expect(wrapper.state()).toEqual({
        search: 'blah',
        searched: true,
        favorites: false,
        errored: false,
      })
    })

    it('should call searchData with the correct paramaters', () => {
      wrapper
        .find('.search-input')
        .simulate('change', { target: { value: 'blah' } })
      wrapper.find('.search-form').simulate('submit', {
        preventDefault: () => {},
      })

      expect(window.fetch).toHaveBeenCalledWith(
        'https://newsapi.org/v2/everything?' +
          'q=' +
          'blah' +
          '&' +
          'language=en&' +
          'sortBy=relevancy&' +
          'apiKey=' +
          apiKey
      )
    })

    it('should call component did mount and match snapshot', () => {
      wrapper.instance().componentDidMount()
      expect(wrapper).toMatchSnapshot()
    })

    it('should call renderArticleCard with search stories', () => {
      const spy = jest.spyOn(wrapper.instance(), 'renderArticleCard')
      wrapper
        .find('.search-input')
        .simulate('change', { target: { value: 'blah' } })
      wrapper.find('.search-form').simulate('submit', {
        preventDefault: () => {},
      })

      expect(spy).toBeCalled()
    })

    it('should call renderArticleCard with top stories', () => {
      const spy = jest.spyOn(wrapper.instance(), 'renderArticleCard')
      wrapper.instance().renderCards()

      expect(spy).toBeCalled()
    })
  })

  describe('mapDispatchToProps', () => {
    it('should call dispatch with saveTopStories', () => {
      const mockDispatch = jest.fn()
      const actionToDispatch = saveTopStories([{ title: 'example' }])

      const mappedProps = mapDispatchToProps(mockDispatch)
      mappedProps.saveTopStories([{ title: 'example' }])

      expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch)
    })

    it('should call dispatch with saveSearchStories', () => {
      const mockDispatch = jest.fn()
      const actionToDispatch = saveSearchStories([{ title: 'example' }])

      const mappedProps = mapDispatchToProps(mockDispatch)
      mappedProps.saveSearchStories([{ title: 'example' }])

      expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch)
    })
  })

  describe('mapStateToProps', () => {
    it('should return an object with topstories and searchstories', () => {
      const mockState = {
        topStories: mockTopStories,
        searchStories: mockSearchStories,
        currentStory: mockCurrentStory,
      }
      const expected = {
        topStories: mockTopStories,
        searchStories: mockSearchStories,
      }
      const mappedProps = mapStateToProps(mockState)

      expect(mappedProps).toEqual(expected)
    })
  })
})
