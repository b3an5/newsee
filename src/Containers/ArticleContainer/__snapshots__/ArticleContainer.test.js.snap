// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArticleContainer articleContainer component should call component did mount and match snapshot 1`] = `ShallowWrapper {}`;

exports[`ArticleContainer articleContainer component should match default snapshot 1`] = `ShallowWrapper {}`;

exports[`ArticleContainer articleContainer component should match snapshot when favorites is clicked 1`] = `ShallowWrapper {}`;

exports[`ArticleContainer articleContainer component should match snapshot when loading 1`] = `ShallowWrapper {}`;

exports[`ArticleContainer articleContainer component should match snapshot when searched 1`] = `ShallowWrapper {}`;

exports[`ArticleContainer articleContainer component should match snapshot when searched retrieves nothing 1`] = `ShallowWrapper {}`;
